python/pyarrow/tests/test_compute.py

```diff
@@ -562,7 +562,7 @@ def test_is_null():
 def test_fill_null():
     arr = pa.array([1, 2, None, 4], type=pa.int8())
     fill_value = pa.array([5], type=pa.int8())
-    with pytest.raises(TypeError):
+    with pytest.raises(pa.ArrowInvalid):
```

Comment:
What did the error message say before, and what does it show now?

Reply:
On master it's
``` python
TypeError: an integer is required (got type pyarrow.lib.Int8Array)
```
verus on this branch
``` python

ArrowInvalid: Could not convert [
  5
] with type pyarrow.lib.Int8Array: tried to convert to int
```

Reply:
Hmm, for this case I find the original error message clearer ..

That's the consequence of the `scalar(..)` conversion using the array conversion under the hood, I suppose? 
But OK, I suppose this is fine (it's maybe mainly the multiline repr of the array in the middle of the sentence that makes it more confusing)